src/smt/theory_str_noodler/formula_preprocess.cpp

```diff
@@ -640,14 +632,14 @@ namespace smt::noodler {
      *
      * @param mn Minimum number of occurrences of a regular sequence to be replaced with a fresh variable.
      */
-    void FormulaPreprocess::reduce_regular_sequence(unsigned mn) {
+    void FormulaPreprocessor::reduce_regular_sequence(unsigned mn) {
         std::map<Concat, unsigned> regs;
         std::set<Predicate> new_eqs;
         get_regular_sublists(regs);
 
         for(const auto& pr : regs) {
             if(pr.second >= mn) {
-                BasicTerm fresh_var = create_fresh_var();
+                BasicTerm fresh_var = util::mk_fresh_noodler_var("regular_seq");
```

Comment:
I think `mk_fresh_var()` (or similar) should not exists anymore. There are 5 functions for creating variables in `util`:
- `mk_int_var_fresh(name)` - creates fresh Z3 `int` variable (by taking name and adding `!NUM` to the end, where `NUM` is unique number handled by Z3)
- `mk_int_var_fresh(name)` - creates fresh Z3 `string` variable similarly as previous
- `mk_fresh_noodler_var(name)` - creates fresh `BasicTerm` variable by adding `!nNUM` to the end, where now `NUM` is handled by us. I do it so that variables with "same" name have the same counter, for debugging (and I add `n` after `!`, so that it does not clashes with name of variables from two previous functions).
- `mk_int_var(name)` - gets Z3 `int` variable with the **exact** name
- `mk_str_var(name)` - similar but `string` variable
The first 2 functions are also in `theory_str_noodler`, they just call `util` functions, we should maybe decide which to keep.

Reply:
It looks great. I consider this issue resolved.